Added a prison on an island

Night-Pryanik opened the pull request:
#### Summary
```SUMMARY: Content "Added a prison on an island."```

#### Purpose of change
[This](https://www.reddit.com/r/cataclysmdda/comments/br9mgo/cast_away_scenario/) Reddit post encouraged me to create an island-based prison. This location could be used as a starting location for Castaway scenario, or a more challenging version of Escape from Prison scenario.

#### Describe the solution
- [x] create an island;
- [x] create a prison;
- [x] fill prison with monsters and items;
- [x] make it a starting location for scenarios.

#### Describe alternatives you've considered
None.

#### Additional context

There is no special or awesome loot, or quests, or NPCs here, at least for now.

Edit: these pics are here only for historic reasons, the more recent pics are in the comments below.

![изображение](https://user-images.githubusercontent.com/11132525/58170566-d4e88600-7ca4-11e9-8ab9-33468365999b.png)
![изображение](https://user-images.githubusercontent.com/11132525/58170580-dd40c100-7ca4-11e9-9eeb-106fd3cdef62.png)

Night-Pryanik commented:
Decided to make it a little bigger - 4x3 instead of a 3x3. Also make my own dock instead of built-in.

Night-Pryanik commented:
Some concrete trails, rocks here and there, slightly adjusted dock, and a helipad.
![изображение](https://user-images.githubusercontent.com/11132525/58184070-71b91c80-7cc1-11e9-825c-5dfd7a5a303b.png)

curstwist commented:
nice additions!

Night-Pryanik commented:
Once again I enlarged the island (I have big plans for it):
![изображение](https://user-images.githubusercontent.com/11132525/58196823-27916480-7cdc-11e9-86ed-e38bf6847da9.png)

Added fence wire and barbed wire - they bordered the high-security zone.

Night-Pryanik commented:
Added part of a prison itself, also added some administrative building.
![изображение](https://user-images.githubusercontent.com/11132525/58204465-64b22280-7ced-11e9-9c68-7565c46bb132.png)

ralreegorganon commented on data/json/mapgen/prison_1.json:
```diff
@@ -259,89 +259,152 @@
+        "1": [ "t_window_bars_alarm" ],
+        "q": [ "t_floor" ],
+        "A": [ "t_floor" ],
+        "`": [
```
For these lists, you can do `[ [ "t_water_dp", 20 ], "t_water_sh" ]` since #28542.

Night-Pryanik replied:
Oh, shoot, I totally forgot about this. Thanks!

Mecares commented:
Great, can't wait to encounter it in a playthrough. Does it have half dug escape tunnel with a spoon as digging tool at the end?

paulenka-aleh commented:
Not until spoon is given digging & mining qualities in game! :)

Night-Pryanik commented:
> Does it have half dug escape tunnel with a spoon as digging tool at the end?

We need to add Rita Hayworth portrait to the game first.

Night-Pryanik commented:
Another update: added officers' house, walking area, shower area, mess hall and a storage, clothes' cleaning area, made infirmary larger, added slight variations to the simple "grass-dirt" landscape. 

![изображение](https://user-images.githubusercontent.com/11132525/58249434-ec407580-7d6f-11e9-84e5-689e47033a04.png)

Night-Pryanik commented:
Close-up view of the prison area
![изображение](https://user-images.githubusercontent.com/11132525/58253988-d3898d00-7d7a-11e9-8a2f-106f720cfed5.png)

The prison itself is mostly done, I just don't know what to place in watchtowers.

Night-Pryanik commented:
Added zombie cops, regular zombies, zombie brutes and z-9s spawn. Anyone else should I add?

Night-Pryanik commented:
I'm not sure if this location could be a good starting location before we mainline Boats mod. Otherwise the only way to escape this island would be swim.

paulenka-aleh commented:
> I'm not sure if this location could be a good starting location before we mainline Boats mod. Otherwise the only way to escape this island would be swim.

Well, first of all you're choosing location, so you should know what you're doing, right?

Secondly it's really good if you're plaing with boats. Alternatively start itself could be put into boats mod...

And finally it gives one more reason to mainline boats thus increasing interest in this task among contributors

Night-Pryanik commented:
> Alternatively start itself could be put into boats mod...

Ok, I'll go this way.

ZhilkinSerg commented:
> 
> 
> > Alternatively start itself could be put into boats mod...
> 
> Ok, I'll go this way.

I think we will mainline boats mod soon.

davidpwbrown commented:
Yep #30816 go ahead :) this looks cool.

Night-Pryanik commented:
Another screenshot to check-up
![изображение](https://user-images.githubusercontent.com/11132525/58368863-c64bda00-7f03-11e9-8e91-2ccbe83887d1.png)

Night-Pryanik commented:
I feel the location is ready to review. The watchtowers, the roof and (possible) second floor could be added in consequent PRs.

ZhilkinSerg commented on data/json/scenarios/scenarios.json:
```diff
@@ -243,6 +243,17 @@
+    "allowed_locs": [ "alcatraz" ],
+    "start_name": "Island prison",
+    "professions": [ "convict", "death_row_convict" ],
+    "flags": [ "LONE_START", "CHALLENGE", "OUTDOOR" ]
```
I am not aware whether `OUTDOOR` flag in scenarios was supported when this was committed, but currently there is no support. You can use `ALLOW_OUTSIDE` for `sloc_prison_alcatraz` though.

Night-Pryanik replied:
Ok, noted.
